### PR TITLE
feat: add tooltip showing info on guardian consensus

### PR DESCRIPTION
**Pull request by isaacknjama**
This PR solves #43 

Before:
image

After (With default value of 5):
image

After (With Leader input):
image

NOTE: Before the Leader fills the `Number of guardians`, a default value of `5` is used for informatory display on the tooltip.

### CODE REVIEW

File: apps/guardian-ui/src/components/ConsensusInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Tooltip } from '@chakra-ui/react';
+import { InfoOutlineIcon } from '@chakra-ui/icons';
+
+export const ConsensusInfo = () => {
+  const totalGuardians = 10;
```
Comment (okjodom) [inline apps/guardian-ui/src/components/ConsensusInfo.tsx:6]: Can we make the example dynamic, based on the current number of guardians selected in the input field?

File: apps/guardian-ui/src/components/ConsensusInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    the expected number of potential malicious guardians is ${potentialMaliciousGuardians}. 
+    This is according to the formula 3*f + 1 = n, 
+    where f is the number of potential malicious guardians, 
+    and n the total number of guardians.
+  `;
+
+  return (
+    <Tooltip
+      label={consensusInfoText}
+      aria-label='Guardian Information'
```
Comment (okjodom) [inline apps/guardian-ui/src/components/ConsensusInfo.tsx:20]: aria labels need to come from translation as well

File: apps/guardian-ui/package.json [hunk 3]
```diff
@@ -36,8 +39,6 @@
     "@types/react-dom": "^18.2.0",
     "@typescript-eslint/eslint-plugin": "^5.48.1",
     "@typescript-eslint/parser": "^5.48.2",
-    "@fedimint/eslint-config": "*",
```
[the remark this answers is not in the document]
Comment (isaack-njama) [reply apps/guardian-ui/package.json:39]: Not really necessary. This happened automatically after adding `@chakra-ui/icons` as a dependency.

**Comment by isaack-njama** [comment]
Updated screenshot on description with recent change.

File: apps/guardian-ui/src/components/ConsensusInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Tooltip } from '@chakra-ui/react';
+import { InfoOutlineIcon } from '@chakra-ui/icons';
+
+export const ConsensusInfo = () => {
+  const totalGuardians = 10;
+  const potentialMaliciousGuardians = 3;
+
+  const consensusInfoText = `
+    If the total number of guardians is ${totalGuardians},
```
[the remark this answers is not in the document]
Comment (isaack-njama) [reply apps/guardian-ui/src/components/ConsensusInfo.tsx:17]: Is there a way to share data between the component and `en.json`?

[the remark this answers is not in the document]
Comment (okjodom) [reply apps/guardian-ui/src/components/ConsensusInfo.tsx:17]: To start with, just hard-code the numbers as part of the string to be translated. Doesn't make sense to have `totalGuardians` and `potentialMaliciousGuardians` as variables if they are not used in a dynamic manner (i.e. popylated from the value the guardian sets as `numPeers`)

File: apps/guardian-ui/src/languages/en.json [hunk 5]
```diff
@@ -56,6 +56,7 @@
     "federation-settings": "Federation settings",
     "federation-name": "Federation name",
     "guardian-number": "Number of guardians",
+    "guardian-information": "Guardian Information",
```
Comment (okjodom) [inline apps/guardian-ui/src/languages/en.json:59]: nit: `guardian-information-aria-label`

File: apps/guardian-ui/src/components/ConsensusInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import { useTranslation } from '@fedimint/utils';
+
+interface Props {
+  numberOfGuardians: string;
+}
+
+export const ConsensusInfo = ({ numberOfGuardians }: Props) => {
+  const { t } = useTranslation();
+
+  const totalGuardians = numberOfGuardians;
```
Comment (okjodom) [inline apps/guardian-ui/src/components/ConsensusInfo.tsx:13]: nit: unnecessary name aliasing

File: apps/guardian-ui/src/components/ConsensusInfo.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (wbobeirne) [reply apps/guardian-ui/src/components/ConsensusInfo.tsx:17]: You can also do this even simpler than the `<Trans />` component, the `t` function takes a second argument of an object with keys to interpolate, so something like:
```json
{
  "my-translation-key": "Interpolation is {{feeling}}"
}
```
```ts
t('my-translation-key', { feeling: "cool" })
```

**Comment by EthnTuttle** [comment]
@isaack-njama Can you get this branch up to date with master? We just discussed this during the [Deep Dive](https://bitcointv.com/w/8MG3h9PN1NLUVX49sWRaxj) today and I will prioritize it's merge once it's more up to date with master.

**Comment by EthnTuttle** [comment]
Closing this PR to revisit again. The UI has changed significantly enough that this could be restart.

**Comment by isaack-njama** [comment]
Hey @EthnTuttle is this PR open for revisit? I'd like to update the changes to match the current UI...

**Pull request closed without merging**